### ISSUE TITLE
Audio/Sound not working over HDMI (KMS) ALSA lib pcm snd_pcm_recover underrun occurred

### ISSUE THREAD

**Issue by shad00m** [midwan/amiberry#990]
**Describe the bug**
When KMS is used, the audio does not work in amiberry when HDMI is selected in the audio settings, only the audio jack works.
HDMI Audio works fine in amiberry when FKMS is selected.

**To Reproduce**
1. set vc4-kms-v3d (default in 64bit bullseye)

2. Let Retropie Compile amiberry/compile it yourself with PLATFORM=rpi4-64-sdl2

3. Run amiberry with any game

4. No HDMI-Audio works

5. error in /dev/shm/runcommand.log
ALSA lib pcm.c:8545:(snd_pcm_recover) underrun occurred

**Expected behavior**
Audio to work in KMS as well not only in FKMS.

- Linux pi 5.15.48-v8+ aarch64
- RetroPie installed through RetroPie-Setup

RetroPie installed 5.2 I think.
I also tried Amiberry by manually compiling it with "PLATFORM=rpi4-64-sdl2"
SDL2 is 2.0.20

**Additional context**
When using KMS I have 3 Audio devices selectable in Amiberry:
```
SDL2: bcm2835 Headphones, bcm2835 Headphones
SDL2: vc4-hdmi-0, MAI PCM i2s-hifi-0
SDL2: vc4-hdmi-1, MAI PCM i2s-hifi-0

```
Only the Headphone jack outputs audio, whenever I select one of the HDMIs the "Sound Emulation" in amiberry gets disabled automatically and the error mentioned above is getting spammed. Setting the config.txt to fkms fixes this problem but fkms causes the mouse cursor to be slow.

**Comment by shad00m** [comment]
> 

thank you for the quick reply
libportaudio2 was already installed unfortunately

**Comment by solskogen** [comment]
Does `speaker-test -c2` produce any sound?

**Comment by solskogen** [comment]
The sound panel shows "SDL2: Built-in Audio Digital Stereo (HDMI)" on my system. You can also try to switch using the other HDMI port.

**Comment by shad00m** [comment]
```
pi@pi:~ $ ps aux |grep -i pulseaudio
pi           738  0.6  1.2 610976 23808 ?        S<sl 11:39   0:00 /usr/bin/pulseaudio --daemonize=no --log-target=journal
```
It appears to be running, however I found: 
```
Jun 21 11:42:17 pi dbus-daemon[357]: [system] Activating via systemd: service name='org.bluez' unit='dbus-org.bluez.service' requested by ':1.113' (uid=1000 pid=1630 comm="/usr/bin/pulseaudio --daemonize=no --log-target=jo")
Jun 21 11:43:35 pi pulseaudio[1630]: Failed to find a working profile.
Jun 21 11:43:35 pi pulseaudio[1630]: Failed to load module "module-alsa-card" (argument: "device_id="2" name="platform-fef05700.hdmi" card_name="alsa_card.platform-fef05700.hdmi" namereg_fail=false tsched=no fixed_latency_range=no ignore_dB=no deferred_volume=yes use_ucm=yes avoid_resampling=no card_properties="module-udev-detect.discovered=1""): initialization failed.

```
Not sure if this is related 



> 

I have tried both HDMI Ports but amiberry switches "sound emulation" to "disabled, but emulated" every time.
https://user-images.githubusercontent.com/9115740/174784385-ad21509d-83ea-4a11-b40e-ab2f52a3cfae.mp4
note: in the video im running it in x11-forwarding just to record the video

**Comment by solskogen** [comment]
Are you running RetroPie on a Bullseye? If so, my guess is that the RetroPie script does something that is not fully compatible with bullseye.

**Comment by shad00m** [comment]
> Are you running RetroPie on a Bullseye? If so, my guess is that the RetroPie script does something that is not fully compatible with bullseye.

yeah, on 64bit and bullseye, which both aren't supported by RetroPie. Oddly enough everything else worked except the audio in amiberry :\
